File: src/main/frontend/src/types/Contract.ts
import { Consumer, consumerDefault } from '../Types'
import { Counter, counterInit } from './Counter'

export interface Contract {
    id: number
    creationDate: number | null
    consumer: Consumer
    counter: Counter
    status: number
    number: number
}

export const ContractInit: Contract = {
    id: 0,
    creationDate: (new Date()).getTime(),
    consumer: consumerDefault,
    counter: counterInit,
    status: -1,
    number: -1
}

File: src/main/java/com/itacon/billing/invoiceservice/model/Contract.java
package com.itacon.billing.invoiceservice.model;

import java.math.BigDecimal;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "contrato")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Contract {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Long id;
	@Column(name = "numero_contrato")
	private BigDecimal number;
	@Column(name = "data_assinatura")
	private Long creationDate;
	@ManyToOne
	@JoinColumn(name = "f_consumidor")
	private Consumer consumer;
	@OneToOne
	@JoinColumn(name = "f_counter")
	private Counter counter;
	@Column(name = "status")
	private Short status;
}
